src/surface/polygon_soup_mesh.cpp

```diff
@@ -10,7 +10,36 @@ namespace geometrycentral {
 
 PolygonSoupMesh::PolygonSoupMesh() {}
 
-PolygonSoupMesh::PolygonSoupMesh(std::string meshFilename) { readMeshFromFile(meshFilename); }
+// TODO, char* can get cast to bool, so using bool and string as optional arguments leads to horrible bugs
+PolygonSoupMesh::PolygonSoupMesh(std::string meshFilename, std::string type) {
+
+  // Attempt to detect filename
+  bool typeGiven = type != "";
+  std::string::size_type sepInd = meshFilename.rfind('.');
+  if (!typeGiven) {
+    if (sepInd != std::string::npos) {
+      std::string extension;
+      extension = meshFilename.substr(sepInd + 1);
+
+      // Convert to all lowercase
+      std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
```

Comment:
Why doesn't this need to be `std::tolower`? Do we have a `using namespace std` somewhere or something? How do namespaces work?!

Reply:
I really don't understand what's going on here, but it only works as `::tolower`, not `std::tolower`. I grabbed this code from `meshio.cpp`, which also doesn't have an `std`. I guess I just don't understand namespaces.

Reply:
Oh good point. I guess I don't understand either. This can remain a mystery for now!